near_deduplication/requirements.txt

```
@@ -0,0 +1,6 @@
+datasets>=2.5.1
```

I think multiprocessing when loading a dataset was introduced in `datasets 2.7` 
```suggestion
datasets>=2.7.0
```